### ISSUE TITLE
Window List Applet not showing LibreOffice documents

### ISSUE THREAD

**Issue by mfreeman72** [linuxmint/cinnamon#10202]
```
 * Cinnamon version: 5.0.4
 * Distribution: Linux Mint 20.2
 * AMD Ryzen 7 APU (also seen on AMD A12 APU), Open Source drivers
 * 64 bit
 ```

**Issue**
After upgrading to LM 20.2 (Cinnamon), the Window-List applet doesn’t work properly with LibreOffice when opening a file by double-clicking on the file within Nemo. All other file types I’ve double-clicked in Nemo and opened in other programs show up in the Window-List applet. When this problem occurs, I can restart cinnamon in the panel menu’s Troubleshooting section, and the documents appear in the Window-List applet after it reloads. This problem did not occur in LM 20.1. This happens in both my laptop as well as my wife’s, both now running 20.2, so it’s not just a one-off thing.

**Steps to reproduce**
Double-click on a LibreOffice file (such as an .odt file) to open it in LO. It will not show up in the Window-List applet (the one in the “Traditional Layout” — I haven’t tried the “Modern Layout” version). However, if opened from "File > Open" or from "File > Recent Documents" in LibreOffice, it *is* shown in Window-List applet, as usual. Again, this seems to be only with LibreOffice. 

**Expected behaviour**
Opening LibreOffice documents by double-clicking should show them in the Window List Applet

**Comment by dajusc** [comment]
Same thing here with LM 20.02, Cinnamon 5.0.4 and LO 6.4.7.2.

It seems to me like the first instance of Writer/Impress/Calc is handled normally. If e.g. one Writer is already open, the second one produces a window-list entry that is colored like a window from another desktop that wants attention. If you click on it, the entry disappears from the window-list.

It doesn't seem to make a difference if writer is launched from the main menu or by double clicking on a file in nemo.

Writer/Impress/Calc aparently do not "affect each other", meaning that one instance of each of them is handled normally.

I tried messing with LO JRE settings (On/Off, JRE11, JRE16) but it didn't seem to do much, except one time I managed to get two writer entries in the window-list.

**Comment by Harry-W-Haines-III** [comment]
Seems related to #10192

**Comment by LuvIsBadToTheBone** [comment]
same with radnom folders or apps like musik folder, video folder, vlc or steam. they just disapear in panel :(

**Comment by mtwebster** [comment]
Pretty sure this is fixed: https://github.com/linuxmint/cinnamon/commit/1c7cc5de912a962f361ed77e47ea082eb95236b1

You can follow these instructions to test it out if you'd like: https://github.com/linuxmint/cinnamon/issues/10209#issuecomment-878484613

**Comment by dajusc** [comment]
@mtwebster https://github.com/linuxmint/cinnamon/issues/10209#issuecomment-878484613 fixes it for me. Thx! :)

**Comment by mtwebster** [comment]
Weird, I don't see that issue. If I double click on an office file from nemo, it opens and comes to the foreground.

**Comment by mfreeman72** [comment]
Just to clarify... The first document opens in the foreground, but any additional documents opened come up in the background.

**Comment by mfreeman72** [comment]
So, if I have a document already opened, and I also open a second document, it is not focused, but opens in the background. Before the 20.2 upgrade, any additional documents open in the foreground, if I'm remembering correctly.

**Comment by mtwebster** [comment]
Sorry I was testing on writer documents (they use the same window) - I can reproduce this with impress.

I'm not arguing that this isn't a regression for cinnamon, but it looks as though what's happening is the intended behavior - take a look at the advanced settings->expert configuration:
![image](https://user-images.githubusercontent.com/262776/125481821-a74bc372-db8c-4003-aa4a-e18f9c10e710.png)

This setting is somewhat explained here: https://github.com/LibreOffice/core/blob/master/officecfg/registry/schema/org/openoffice/Office/Common.xcs#L2839-L2850

Setting that to true (making sure to ok->ok out of all of the options windows) mostly (for some reason not entirely) fixes the issue.

I went back to a virtual machine that had 20.1 and libreoffice 6.4.6 and the new window was focused every time. I then upgraded it to 6.4.7.2 (which mint 20.2 has) and began having focus issues, so it seems a bit more muddy than I thought.

For the time being you can do:
```
gsettings set org.cinnamon demands-attention-passthru-wm-classes "['gnome-screenshot', 'lxterminal', 'xfce4-terminal', 'firefox', 'libreoffice']"
```
Adding to that setting effectively restores cinnamon 4.8 behavior when launching programs. But it doesn't work 100% of the time, and I'm not sure what the reason for that is, other than blaming libreoffice.

**Comment by mfreeman72** [comment]
Odd. LibreOffice Writer has always opened documents in separate windows on every system I've used, not the same window. How do you activate that feature? Does it use tabs or something like that? I've never seen that.

I'll try out those settings and see what they do. Thanks!

**Comment by mfreeman72** [comment]
Also, I've been using LO 7.1.4 (from their official PPA) in both LM 20.1 and 20.2. It didn't have this behavior in 20.1, but does in 20.2. So I'm assuming the difference must be in the Cinnamon version and not the LibreOffice version.

**Comment by mfreeman72** [comment]
Ok. I've tried both the Expert Configuration setting in LibreOffice, as well as the gsettings command. Neither work. LibreOffice Writer documents are still opening in the background instead of the foreground. I'm still convinced this is a Cinnamon issue, since it only occurs in the newer version of Cinnamon and not the old version, when using the same version of LibreOffice.

**Comment by mtwebster** [comment]
Ok, it looks like libreoffice doesn't set an identifier (wm_class) until after cinnamon first sees the window. When it starts, it's identified as 'soffice', then later changes to 'libreoffice-impress' or whatever specific app it is.

Try having *both* `libreoffice` and `soffice` in that setting

**Comment by mfreeman72** [comment]
Yes. Adding soffice to the gsettings command seems to fix the problem. Thank you!

**Comment by mtwebster** [comment]
I was wrong about writer and multiple windows btw - I had accidentally slipped the broken window-list applet back into my system, so new windows were just disappearing (and since I was testing copies of the same document, I missed the difference).

**Comment by Glint-Eye** [comment]
I experienced the same "new window doesnt get focus" behaviour with nemo when i started nemo through a custom launcher (gnome-pie). i added nemo to the exception list. works fine now.

```
gsettings set org.cinnamon demands-attention-passthru-wm-classes "['gnome-screenshot', 'lxterminal', 'xfce4-terminal', 'firefox', 'libreoffice', 'soffice', 'nemo']"
```

**Comment by mtwebster** [comment]
@Glint-Eye what version of cinnamon are you running?

**Comment by Glint-Eye** [comment]
I'm on 5.0.5

**Comment by mfreeman72** [comment]
The above workaround solves the issue.
